src/document_basic.c

```diff
@@ -173,24 +176,19 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx) {
     FieldSpec *field = &spec->fields[ii];
 
     jsonIter = japi->get(jsonRoot, field->path);
-    if (!jsonIter) {
+    // if field does not exist or is empty (can happen after JSON.DEL)
+    if (!jsonIter || japi->len(jsonIter) == 0) {
         continue;
     }
 
-    RedisJSON jsonValue = japi->next(jsonIter);
-    if (!jsonValue || 
-        FieldSpec_CheckJsonType(field->types, japi->getType(jsonValue)) != REDISMODULE_OK) {
-      goto done;
-    }
-
     size_t oix = doc->numFields++;
     doc->fields[oix].path = rm_strdup(field->path);
     doc->fields[oix].name = (field->name == field->path) ? doc->fields[oix].path
                                                          : rm_strdup(field->name);
 
     // on crdt the return value might be the underline value, we must copy it!!!
     // TODO: change `fs->text` to support hash or json not RedisModuleString
-    if (JSON_GetRedisModuleString(ctx, jsonValue, &doc->fields[oix].text) != REDISMODULE_OK) {
+    if (JSON_GetRedisModuleString(ctx, jsonIter, field->types, &doc->fields[oix]) != REDISMODULE_OK) {
```

Comment:
We would get here always now, and do the above `rm_strdup`, but before this change,
`FieldSpec_CheckJsonType` was called before instead of being called from within `JSON_GetRedisModuleString`
And the above allocations were avoided.
I assume you can consider this when addressing the [comment](https://github.com/RediSearch/RediSearch/pull/2133/files#r668482861) from Meir